Blish HUD/GameServices/GameIntegration/AudioIntegration.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Blish_HUD.Debug;
+using CSCore.CoreAudioAPI;
+using Microsoft.Xna.Framework;
+
+namespace Blish_HUD.GameIntegration {
+    public sealed class AudioIntegration : IUpdatable, IDisposable {
+
+        private static readonly Logger Logger = Logger.GetLogger<AudioIntegration>();
+
+        private const int   CHECK_INTERVAL     = 250;
+        private const int   AUDIOBUFFER_LENGTH = 20;
+        private const float MAX_VOLUME         = 0.4f;
+
+        private readonly GameIntegrationService _gameIntegration;
+        private readonly MMDeviceEnumerator     _deviceEnumerator;
+        private readonly RingBuffer<float>      _audioPeakBuffer = new RingBuffer<float>(AUDIOBUFFER_LENGTH);
+
+        private AudioSessionManager2  _activeAudioDeviceManager;
+        private AudioMeterInformation _processMeterInformation;
+
+        private double _timeSinceCheck = 0;
+
+        public float AverageGameVolume {
+            get {
+                float total = 0;
+                for (int i = 0; i < _audioPeakBuffer.InternalBuffer.Length; i++) {
+                    total += _audioPeakBuffer.InternalBuffer[i];
+                }
+
+                return MathHelper.Clamp(total / _audioPeakBuffer.InternalBuffer.Length, 0, MAX_VOLUME);
+            }
+        }
```

Comment:
I would only let this compute the average if there is a need. So have a backing field that gets populated if null.
You could reset it to null at line 83. Especially in debug runs where this is called each frame, we could save some cycles 🙂